external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -75,1474 +80,804 @@
-                            void * userData)
+void *lwm2m_connect_server(uint16_t secObjInstID,
+						   void *userData)
 {
-  client_data_t * dataP;
-  lwm2m_list_t * instance;
-  dtls_connection_t * newConnP = NULL;
-  dataP = (client_data_t *)userData;
-  lwm2m_object_t  * securityObj = dataP->securityObjP;
-
-  instance = LWM2M_LIST_FIND(dataP->securityObjP->instanceList, secObjInstID);
-  if (instance == NULL) return NULL;
-
-  newConnP = connection_create(dataP->connList, dataP->sock, securityObj, instance->id, dataP->lwm2mH, dataP->addressFamily);
-  if (newConnP == NULL)
-  {
-      fprintf(stderr, "Connection creation failed.\n");
-      return NULL;
-  }
-
-  dataP->connList = newConnP;
-  return (void *)newConnP;
+	client_data_t *dataP;
+	lwm2m_list_t *instance;
+	dtls_connection_t *newConnP = NULL;
+	dataP = (client_data_t *)userData;
+	lwm2m_object_t   *securityObj = dataP->securityObjP;
+
+	instance = LWM2M_LIST_FIND(dataP->securityObjP->instanceList, secObjInstID);
+	if (instance == NULL) {
+		return NULL;
+	}
+
+
+	newConnP = connection_create(dataP->connList, dataP->sock, securityObj, instance->id, dataP->lwm2mH, dataP->addressFamily);
+	if (newConnP == NULL) {
+		fprintf(stderr, "Connection creation failed.\n");
+		return NULL;
+	}
+
+	dataP->connList = newConnP;
+	return (void *)newConnP;
 }
 #else
-void * lwm2m_connect_server(uint16_t secObjInstID,
-                            void * userData)
+void *lwm2m_connect_server(uint16_t secObjInstID,
+						   void *userData)
 {
-    client_data_t * dataP;
-    char * uri;
-    char * host;
-    char * port;
-    connection_t * newConnP = NULL;
-    coap_protocol_t proto = COAP_UDP;
+	client_data_t *dataP;
+	char *uri;
+	char *host;
+	char *port;
+	connection_t *newConnP = NULL;
 
-    dataP = (client_data_t *)userData;
+	dataP = (client_data_t *)userData;
 
-    uri = get_server_uri(dataP->securityObjP, secObjInstID);
+	uri = get_server_uri(dataP->securityObjP, secObjInstID);
 
-    if (uri == NULL) return NULL;
+	if (uri == NULL) {
+		return NULL;
+	}
 
-    // parse uri in the form "coaps://[host]:[port]"
+	// parse uri in the form "coaps://[host]:[port]"
+	if (0 == strncmp(uri, "coaps://", strlen("coaps://"))) {
+		host = uri + strlen("coaps://");
+	} else if (0 == strncmp(uri, "coap://",  strlen("coap://"))) {
+		host = uri + strlen("coap://");
+	} else {
+		goto exit;
+	}
+	port = strrchr(host, ':');
+	if (port == NULL) {
+		goto exit;
+	}
+	// remove brackets
+	if (host[0] == '[') {
+		host++;
+		if (*(port - 1) == ']') {
+			*(port - 1) = 0;
+		} else {
+			goto exit;
+		}
+	}
+	// split strings
+	*port = 0;
+	port++;
 
-    proto = coap_get_protocol_from_uri(uri);
-    if (proto >= COAP_PROTOCOL_MAX) {
-        fprintf(stderr, "Not supported protocol : %d\n", proto);
-        goto exit;
+	fprintf(stderr, "Opening connection to server at %s:%s\r\n", host, port);
+	newConnP = connection_create(dataP->connList, dataP->sock, host, port, dataP->addressFamily);
```

Comment:
Below code is a latest one.

fprintf(stderr, "Opening connection to server at %s:%s\r\n", host, port);
newConnP = connection_create(COAP_UDP, dataP->connList, dataP->sock, host, port, dataP->addressFamily);